python/dllib/src/bigdl/dllib/nncontext.py

```diff
@@ -409,7 +405,121 @@ def _read_stream(fd, fn):
+       In this case, please set the Spark configurations through command line options or
+       the properties file. You need to use "spark-submit" for yarn-cluster or k8s-cluster mode.
+       To make things easier, you are recommended to use the launch scripts we provide:
+       https://github.com/intel-analytics/analytics-zoo/tree/master/scripts.
+
+       For other cluster modes, you are recommended to install and run analytics-zoo through
+       pip, which is more convenient.
+    :param spark_log_level: The log level for Spark. Default to be 'WARN'.
+    :param redirect_spark_log: Whether to redirect the Spark log to local file. Default to be True.
+
+    :return: An instance of SparkContext.
+    """
+    cluster_mode = cluster_mode.lower()
+    memory = "2g"
+    cores = 2
+    num_nodes = 1
+
+    spark_args = {}
+    spark_args["spark_log_level"] = spark_log_level
+    spark_args["redirect_spark_log"] = redirect_spark_log
+    if not isinstance(conf, six.string_types):
+        memory = conf.get("spark.executor.memory", "2g")
+        cores = conf.get("spark.executor.cores", 2)
+        num_nodes = conf.get("spark.executor.instances", 1)
+        spark_args.update(conf.getAll())
+    if cluster_mode == "spark-submit":
+        sc = init_internal_nncontext(conf, spark_log_level, redirect_spark_log)
+    elif cluster_mode == "local":
+        os.environ["SPARK_DRIVER_MEMORY"] = memory
+
+        if "python_location" in kwargs:
+            spark_args["python_location"] = kwargs["python_location"]
+        sc = init_spark_on_local(cores, **spark_args)
+    elif cluster_mode in ("yarn-client", "yarn-cluster"):  # yarn-cluster or yarn-client
+        hadoop_conf = os.environ.get("HADOOP_CONF_DIR")
+        if not hadoop_conf:
+            assert "hadoop_conf" in kwargs, \
+                "Directory path to hadoop conf not found for yarn-client mode. Please either " \
+                "specify argument hadoop_conf or set the environment variable HADOOP_CONF_DIR"
+            hadoop_conf = kwargs["hadoop_conf"]
+        from bigdl.dllib.utils.utils import detect_conda_env_name
+
+        conda_env_name = detect_conda_env_name()
+        for key in ["driver_cores", "driver_memory", "extra_executor_memory_for_ray",
+                    "extra_python_lib", "penv_archive", "additional_archive",
+                    "hadoop_user_name", "spark_yarn_archive", "jars"]:
+            if key in kwargs:
+                spark_args[key] = kwargs[key]
+        if cluster_mode == "yarn-client":
+            from bigdl.dllib.nncontext import init_spark_on_yarn
+            sc = init_spark_on_yarn(hadoop_conf=hadoop_conf,
+                                    conda_name=conda_env_name,
+                                    num_executors=num_nodes, executor_cores=cores,
+                                    executor_memory=memory, **spark_args)
+        else:
+            if os.environ.get("OnAppMaster", "False") == "True":
```

Comment:
Why move the if here？`init_spark_on_yarn_cluster` won't be an independent function.